### DISCUSSION TITLE
How to set a Page description

### DISCUSSION THREAD

**Discussion by DennisKuhn** [pnp/powershell#657] [Q&A] [answered]
Hello,
when I create a page using 
`$page = Add-PnPPage -Name $name -PromoteAs NewsArticle -CommentsEnabled`
and set author using:
```$page.PageHeader.Authors = "[{`"id`":`"$($user.LoginName)`"}]";
$page.PageHeader.AuthorByLine = "[`"$($user.Email)`"]";```
and add content using:
`Add-PnPPageSection -Page $page -SectionTemplate OneColumn`
`Add-PnPPageTextPart -Page $page -Text $mainTextValue`
and set title and header image using:
`Set-PnPPage -Identity $name -Title $titleValue`
`Set-PnPPage -Identity $name -HeaderType Custom -ServerRelativeImageUrl $HeaderImage -TranslateX 10.5 -TranslateY 11.0`

The description contains the html code of the first text block (`$mainTextValue`). The page shows correctly, e.g. contains HTML formatting. The description shown in preview shows some HTML code. **I wonder how to set the description using PNP PowerShell?** Any hinters are much appreciated. I've tried setting the field "Description" on the List Item in SitePages without success.

The content of $mainTextValue is:
```
<html><head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"><meta content="text/html; charset=utf-8"><style type="text/css" style="display:none">
<!--
p
	{margin-top:0;
	margin-bottom:0}
-->
</style></head><body dir="ltr"><div style="font-family:Calibri,Arial,Helvetica,sans-serif; font-size:12pt; color:rgb(0,0,0); background-color:rgb(255,255,255)">Test <b>Post</b> hoping for a description</div><div dir="ltr" style="direction:ltr"></div></body></html>
```

![Code in News Preview](https://user-images.githubusercontent.com/992724/117224113-48b8d480-ae63-11eb-89b3-b24c9b8d6d4b.png)

![Code in News Preview PAge](https://user-images.githubusercontent.com/992724/117224394-ee6c4380-ae63-11eb-95d9-dd5626e82b72.png)

**Comment by LeonArmston** [comment, marked as answer]
Hi Dennis

I've just been able to change the description in my demo environment by altering the list item description using Set-PnPListItem (see my screenshot below)

![image](https://user-images.githubusercontent.com/12968962/117269001-cb379900-ae4f-11eb-9b85-585401bff655.png)

Does this work for you?

Thanks

Leon

**Reply by DennisKuhn** [reply]
Awesomest, thank you that worked. I tried it before but failed because I specified the list wrong.
Instead of 
`$reply = Set-PnPListItem -List $($page.PagesLibrary) -Id $($page.PageId) -Values @{ "Description" = "Hi All" }`
I had to use
`$reply = Set-PnPListItem -List $($page.PagesLibrary.Id) -Id $($page.PageId) -Values @{ "Description" = "Hi All" }`
